cloud_utils/cache/stores/pickle_store.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import os
+import pickle
+import time
+from typing import Any, Callable, Dict, Tuple
+
+import gamla
+
+
+def _local_cache_path(cache_name: str, path: str):
+    return os.path.join(path, f"{cache_name}.pickle")
+
+
+@gamla.timeit
+def _load_cache_from_local(cache_name: str, path: str) -> Dict[Tuple, Any]:
+    with open(_local_cache_path(cache_name, path), "rb") as local_cache_file:
+        return pickle.load(local_cache_file)
+
+
+def _save_cache_locally(cache_name: str, path: str, cache: Dict[Tuple, Any]):
+    if not os.path.exists(path):
+        os.makedirs(path)
+    local_path = _local_cache_path(cache_name, path)
+    with open(local_path, "wb") as local_cache_file:
+        pickle.dump(cache, local_cache_file)
+    logging.info(f"Saved {len(cache)} cache items locally for {cache_name}.")
+
+
+def make_store(name: str, cache_path: str) -> Tuple[Callable, Callable]:
+    change_count = 0
+    # Initialize cache.
+    try:
+        cache = _load_cache_from_local(name, cache_path)
+        logging.info(f"Loaded {len(cache):,} cache items from local file for {name}.")
+    except (OSError, IOError, EOFError, pickle.UnpicklingError) as err:
+        logging.info(
+            f"Cache {name} does not exist or is invalid. Initializing an empty cache.",
+        )
+        logging.error(err)
+        cache = {}
+
+    def get_item(key: Tuple):
+        return cache[key]
+
+    def set_item(key: Tuple, value):
+        nonlocal change_count
+
+        change_count += 1
+        cache[key] = value
+
+        if change_count >= 10:
+            logging.info(
+                f"More than 10 keys changed in cache {name}. Syncing with local file.",
+            )
+
+            try:
+                start_time = time.time()
+                _save_cache_locally(name, cache_path, cache)
+                end_time = time.time()
+                logging.info(
```

Comment:
i think we can remove this and the time checks, it was an experiment to see if it slows things down